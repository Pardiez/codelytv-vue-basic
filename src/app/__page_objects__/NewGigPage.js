import PageObject from './PageObject'

export default class NewGigPage extends PageObject {
  constructor(wrapper) {
    super(wrapper)
    this.wrapper = wrapper
  }

  writeName(name) {

  }
}
