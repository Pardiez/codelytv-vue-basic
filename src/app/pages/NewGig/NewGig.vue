<template>
  <div>
    <e-input
      label="Name"
      v-model="name"
      :validation="$v.name"
      :validation-messages="validationMessages.name"
    ></e-input>
    <e-date-time
      format="YYYY/MM/DD - hh:mm"
      label="Date and time"
      v-model="dateTime"
      :validation="$v.dateTime"
      :validation-messages="validationMessages.dateTime"
      :value="dateTime"
    ></e-date-time>
  </div>
</template>

<script>
  import { required, minLength, maxLength } from 'vuelidate/lib/validators'
  import eInput from './embeeded/eInput.vue'
  import eDateTime from './embeeded/eDateTime.vue'
  import { isFutureDatetime } from './customValidations'

  export default {
    props: {
    },
    validations: {
      name: {
        required,
        minLength: minLength(5),
        maxLength: maxLength(20)
      },
      dateTime: {
        required,
        isFutureDatetime
      }
    },
    data () {
      return {
        name: '',
        dateTime: '',
        validationMessages: {
          name: {
            required: 'Name is required.',
            minLength: 'Minimum 5 characters.',
            maxLength: 'Maximum 20 characters.'
          },
          dateTime: {
            required: 'Date of birth is required.',
            isFutureDatetime: 'You cannot set a gig in a past date :('
          }
        }
      }
    },
    components: {
      eInput, eDateTime
    }
  }
</script>
