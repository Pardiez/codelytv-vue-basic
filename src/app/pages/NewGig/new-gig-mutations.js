export const CREATE_GIG_REQUEST = 'createGigRequest'
export const CREATE_GIG_SUCCESS = 'createGigSuccess'
export const CREATE_GIG_ERROR = 'createGigError'

export function createGigRequest(state) {
  state.loading = true
  state.error = ''
}

export function createGigSuccess(state, gig) {
  state.loading = false
  const newGig = { name: gig.title, datetime: gig.day }
  const newDay = {day: gig.day, gigs: [newGig]}
  state.days.push(newDay)
  state.error = ''
}

export function createGigError(state, error) {
  state.loading = false
  state.error = error
}
