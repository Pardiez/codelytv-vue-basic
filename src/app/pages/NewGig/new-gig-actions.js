import { createGig } from '../../shared/mosica-api'
import { CREATE_GIG_ERROR, CREATE_GIG_REQUEST, CREATE_GIG_SUCCESS } from './new-gig-mutations'

export function buildCreateGigAction() {
  return createGigAction(createGig).run
}

export function createGigAction(createGig) {
  return { run }

  async function run({ commit }, params) {
    commit(CREATE_GIG_REQUEST)
    try {
      await createGig(params)
      commit(CREATE_GIG_SUCCESS, params)
    }
    catch (error) {
      commit(CREATE_GIG_ERROR, error)
    }
  }
}
