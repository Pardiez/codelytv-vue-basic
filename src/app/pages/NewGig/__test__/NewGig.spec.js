import { mount } from 'vue-test-utils'
import NewGig from '@/app/pages/NewGig/NewGig.vue'
import Vue from 'vue'
import Vuelidate from 'vuelidate'
import Quasar from 'quasar-framework'
import NewGigPage from '../../../__page_objects__/NewGigPage'
Vue.use(Vuelidate)
Vue.use(Quasar)

describe('New Gig', () => {

  let page, wrapper
  beforeEach(async () => {
    wrapper = mount(NewGig)
    page = new NewGigPage(wrapper)
  })

  it('matches full snapshot', async () => {
    page.matchSnapshot()
  })

  it('when validating name', async () => {
    console.log(page.text())

    page.writeName('abc')
    expect(page.text()).toContain('Minimum 5 characters.')
  })
})
