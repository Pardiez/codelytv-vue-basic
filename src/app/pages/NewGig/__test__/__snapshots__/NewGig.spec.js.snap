// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`New Gig matches full snapshot 1`] = `
<div>
    <div value="">
        <div><label>Name</label>
            <div class="q-if row no-wrap items-center relative-position q-input text-primary" required="required">
                <!---->
                <div class="q-if-inner col row no-wrap items-center relative-position">
                    <!---->
                    <!----><input type="text" class="col q-input-target text-left">
                    <!---->
                    <!---->
                    <!---->
                    <!---->
                </div>
                <!---->
            </div> <span><span class="label text-red" style="display: none;">

        Name is required.
      </span><span class="label text-red" style="display: none;">

        Minimum 5 characters.
      </span><span class="label text-red" style="display: none;">

        Maximum 20 characters.
      </span></span>
        </div>
    </div>
    <div value=""><label>Date and time</label>
        <div tabindex="0" class="q-if row no-wrap items-center relative-position q-datetime-input q-if-focusable text-primary" label="Date and time">
            <!---->
            <div class="q-if-inner col row no-wrap items-center relative-position">
                <!---->
                <!---->
                <div class="col row items-center q-input-target justify-start"></div>
                <!---->
            </div> <i aria-hidden="true" class="q-if-control q-icon material-icons">arrow_drop_down</i>
            <!---->
        </div> <span><span class="label text-red" style="display: none;">
      Date of birth is required.
    </span><span class="label text-red" style="display: none;">
      You cannot set a gig in a past date :(
    </span></span>
    </div>
</div>
`;
